specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2022-09-01/storagesync.json

```diff
@@ -0,0 +1,5992 @@
+        },
+        "agentVersionExpirationDate": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Registered Server Agent Version Expiration Date",
+          "readOnly": true
+        },
+        "serverOSVersion": {
+          "type": "string",
+          "description": "Registered Server OS Version"
+        },
+        "serverManagementErrorCode": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Registered Server Management Error Code"
+        },
+        "lastHeartBeat": {
+          "type": "string",
+          "description": "Registered Server last heart beat"
+        },
+        "provisioningState": {
+          "type": "string",
+          "description": "Registered Server Provisioning State"
+        },
+        "serverRole": {
+          "type": "string",
+          "description": "Registered Server serverRole"
+        },
+        "clusterId": {
+          "type": "string",
+          "description": "Registered Server clusterId"
+        },
+        "clusterName": {
+          "type": "string",
+          "description": "Registered Server clusterName"
+        },
+        "serverId": {
+          "type": "string",
+          "description": "Registered Server serverId"
+        },
+        "storageSyncServiceUid": {
+          "type": "string",
+          "description": "Registered Server storageSyncServiceUid"
+        },
+        "lastWorkflowId": {
+          "type": "string",
+          "description": "Registered Server lastWorkflowId"
+        },
+        "lastOperationName": {
+          "type": "string",
+          "description": "Resource Last Operation Name"
+        },
+        "discoveryEndpointUri": {
+          "type": "string",
+          "description": "Resource discoveryEndpointUri"
+        },
+        "resourceLocation": {
+          "type": "string",
+          "description": "Resource Location"
+        },
+        "serviceLocation": {
+          "type": "string",
+          "description": "Service Location"
+        },
+        "friendlyName": {
+          "type": "string",
+          "description": "Friendly Name"
+        },
+        "managementEndpointUri": {
+          "type": "string",
+          "description": "Management Endpoint Uri"
+        },
+        "monitoringEndpointUri": {
+          "type": "string",
+          "description": "Telemetry Endpoint Uri"
+        },
+        "monitoringConfiguration": {
+          "type": "string",
+          "description": "Monitoring Configuration"
+        },
+        "serverName": {
+          "type": "string",
+          "description": "Server name",
+          "readOnly": true
+        },
+        "applicationId": {
+          "type": "string",
+          "description": "Server Application Id"
+        },
+        "identity": {
+          "type": "boolean",
+          "description": "Apply server with newly discovered ApplicationId if available.",
+          "readOnly": true
```

Comment:
why readOnly?

Reply:
It is RegisteredServerProperties , once it is set , It cannot be reset.